src/guides/v2.4/ext-best-practices/phpstorm/built-in-upgrade-compatibility-tool.md

```diff
@@ -0,0 +1,92 @@
+---
+group: extension-best-practices
+title: The built-in Upgrade Compatibility Tool (MVP)
+functional_areas:
+- Standards
+---
+
+The built-in Upgrade Compatibility Tool aims to achieve the same issues coverage as original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html).
+There are issues that covered by this MVP version:
+
+| Code | Name |
+| ----- | ------ |
+| 1131 | extending from @deprecated class |
+| 1132 | importing @deprecated class |
+| 1332 | importing @deprecated interface |
+| 1134 | using @deprecated class |
+| 1334 | using @deprecated interface |
+| 1234 | using @deprecated constant |
+| 1534 | using @deprecated property |
+| 1235 | overriding @deprecated constant |
+| 1535 | overriding @deprecated property |
+| 1337 | inherited from @deprecated interface |
+| 1338 | implemented @deprecated interface |
+| 1439 | call @deprecated method |
+
+Complete list of inspections that are already available in the original tool can be found by the [following link]({{site.baseurl}}/upgrade-compatibility-tool/errors.html).
+
+This feature can be used in three different ways from the user's point of view:
+
+*  run compatibility inspections in a [run tool window](https://www.jetbrains.com/help/idea/run-tool-window.html) with the familiar output as in the original Adobe Commerce [Upgrade Compatibility Tool]({{site.baseurl}}/upgrade-compatibility-tool/introduction.html)
+*  run inspections in a real time in the files that are opened in the editor (default)
+*  [run inspections manually](https://www.jetbrains.com/help/idea/running-inspections.html) through the user interface on the selected scope of files (default)
+
+It was developed with an aim to use IntelliJ IDEA in most effective way. So, inspections that are running during
+the code analysis can alternatively be enabled in the [inspections settings](https://www.jetbrains.com/help/idea/code-inspection.html)
+`Preferences -> Editor -> Inspections -> UCT`:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-settings.png)
+
+It allows you to see the problematic code just in real time without running it intentionally during code writing or even files visiting.
+**The result of the real time inspection is most suitable for using during development to be always compatible with the future versions**:
+
+![]({{site.baseurl}}/common/images/phpstorm/built-in-uct/uct-inspections-highlighting.png)
+
+The second cool IntelliJ IDEA tool under which we adapted our code by reusing **Inspections** feature of it is running inspections through special action
```

Comment:
Line 45 - 46 : The English grammar used is very mixed up. Please rewrite in plainer English and use more than one sentence. Omit the word "cool"

I tried to correct the syntax but wasn't totally sure of the meaning of the two sentences.

Reply:
Done.